src/components/Input/Checkbox.tsx

```diff
@@ -27,13 +30,14 @@ function Checkbox({ className, id, intermediate, isChecked, justify, label, labe
         `${className}`,
         'cursor-pointer flex items-center transition ease-in-out',
         focused && 'ring-2 ring-panel-icon-action ring-inset',
+        disabled && 'opacity-50 cursor-auto',
       ])}
     >
       <input
         id={id}
         type="checkbox"
-        checked={isChecked}
-        onChange={onChange}
+        checked={!disabled && isChecked}
```

Comment:
There may be a possibility of a checked checkbox being disabled somewhere.

Reply:
Only if the ``disabled`` prop is changed to true since its optional and always has an initial value of false.